modules/nextflow/src/main/groovy/nextflow/script/ScriptRunner.groovy

```diff
@@ -225,8 +225,11 @@ class ScriptRunner {
 
     protected void parseScript( ScriptFile scriptFile, String entryName ) {
         scriptParser = new ScriptParser(session)
-                            .setEntryName(entryName)
-                            .parse(scriptFile.main)
+        scriptParser.setEntryName(entryName)
+        // don't execute entry workflow if preview action (i.e. inspect command) is specified
+        if( previewAction )
+            scriptParser.setModule(true)
```

Comment:
Since the previewAction was created for the inspect command and isn't used for anything else, I use it here to disable the entry workflow execution

Setting the module flag will register the process definitions without running the entry workflow: https://github.com/nextflow-io/nextflow/blob/53d6c61ff39464b42a3b77ab0cbd61328575db77/modules/nextflow/src/main/groovy/nextflow/script/BaseScript.groovy#L159-L162

Reply:
Then I'd say to get rid of `previewAction` (since it's useless) and use instead [ContainerInspectMode](https://github.com/nextflow-io/nextflow/blob/improve-inspect-command/modules/nextflow/src/main/groovy/nextflow/container/inspect/ContainerInspectMode.groovy#L31-L31) that's more explicit about the intent.

Reply:
Ok, I was understanding that `previewAction` was not used anymore, but it's still needed 👍

Reply:
I think we could remove the previewAction by refactoring CmdInspect to parse and run the script directly instead of going through CmdRun. Which might be possible since we probably don't need so much run logic with the new approach. Something to consider in a future iteration